### ISSUE TITLE
MagiskManager redownloads Magisk.zip

### ISSUE THREAD

**Issue by aapo** [topjohnwu/Magisk#515]
MagiskManager-v5.8.3
Even I have correct file /sdcard/Magisk-v16.0(1600).zip MagiskManager wants to download it.
When I press Install, Magisk will ask 

    Do you want to install Magisk-v16.0(1600).zip now?
    no thanks   /    release notes    / install

It will redownload zip even I have zip already unzipped:

    ls /data/user/0/com.topjohnwu.magisk/install/                                  <
    boot_patch.sh
    busybox
    chromeos
    magisk
    magiskboot
    magiskinit
    new-boot.img
    stock_boot_c17bef9465c8f128b99a824d7fb4f96b0b9f71ed.img.gz
    util_functions.sh

**Comment by neekless** [comment]
This is intended behavior. It will always redownload the zip.

A future update will check for the zip's MD5 and prevent redownloading if it matches ([this commit](https://github.com/topjohnwu/Magisk/commit/9f05b182a260258c60eead9dccaba245b74cd81b)), but this update has not been pushed out yet.

**Comment by topjohnwu** [comment]
@aapo just as @neekless mentioned, it is already added and this feature will soon be available.

**Comment by aapo** [comment]
Great, but I can't get it working. Even this simple test failed:
I pulled git head, compiled MagiskManager.apk and installed it to the device
Then I put correct file to the correct position, using:

    wget https://github.com/topjohnwu/Magisk/releases/download/v16.0/Magisk-v16.0.zip
    adb push Magisk-v16.0.zip "/sdcard/Download/Magisk-v16.0(1600).zip" 

Then I started application, it check updates and found v16.0 and starts to downloading.

I also test with commit https://github.com/topjohnwu/Magisk/commit/9f05b182a260258c60eead9dccaba245b74cd81b and it failed the same way.
